### PR TITLE
Use unique table name for TestKuduIntegrationDecimalColumns

### LINKED ISSUE #7317: Flaky test TestKuduIntegrationDecimalColumns. testDeleteByPrimaryKeyDecimalColumn
https://github.com/trinodb/trino/runs/2129444090
```
Error:  Tests run: 227, Failures: 1, Errors: 0, Skipped: 32, Time elapsed: 679.51 s <<< FAILURE! - in TestSuite
Error:  testDeleteByPrimaryKeyDecimalColumn(io.trino.plugin.kudu.TestKuduIntegrationDecimalColumns)  Time elapsed: 0.507 s  <<< FAILURE!
java.lang.RuntimeException: line 1:32: Column 'decimal_id' cannot be resolved
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:120)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:491)
	at io.trino.testing.QueryAssertions.assertUpdate(QueryAssertions.java:71)
	at io.trino.testing.AbstractTestQueryFramework.assertUpdate(AbstractTestQueryFramework.java:235)
	at io.trino.testing.AbstractTestQueryFramework.assertUpdate(AbstractTestQueryFramework.java:230)
	at io.trino.plugin.kudu.TestKuduIntegrationDecimalColumns.testDeleteByPrimaryKeyDecimalColumn(TestKuduIntegrationDecimalColumns.java:95)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: io.trino.spi.TrinoException: line 1:32: Column 'decimal_id' cannot be resolved
	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:48)
	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:43)
	at io.trino.sql.analyzer.SemanticExceptions.missingAttributeException(SemanticExceptions.java:33)
	at io.trino.sql.analyzer.Scope.lambda$resolveField$7(Scope.java:228)
	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
	at io.trino.sql.analyzer.Scope.resolveField(Scope.java:228)
	at io.trino.sql.analyzer.ExpressionAnalyzer$Visitor.visitIdentifier(ExpressionAnalyzer.java:485)
	at io.trino.sql.analyzer.ExpressionAnalyzer$Visitor.visitIdentifier(ExpressionAnalyzer.java:404)
	at io.trino.sql.tree.Identifier.accept(Identifier.java:72)
	at io.trino.sql.tree.StackableAstVisitor.process(StackableAstVisitor.java:27)
	at io.trino.sql.analyzer.ExpressionAnalyzer$Visitor.process(ExpressionAnalyzer.java:427)
	at io.trino.sql.analyzer.ExpressionAnalyzer$Visitor.getOperator(ExpressionAnalyzer.java:1750)
	at io.trino.sql.analyzer.ExpressionAnalyzer$Visitor.visitComparisonExpression(ExpressionAnalyzer.java:600)
	at io.trino.sql.analyzer.ExpressionAnalyzer$Visitor.visitComparisonExpression(ExpressionAnalyzer.java:404)
	at io.trino.sql.tree.ComparisonExpression.accept(ComparisonExpression.java:71)
	at io.trino.sql.tree.StackableAstVisitor.process(StackableAstVisitor.java:27)
	at io.trino.sql.analyzer.ExpressionAnalyzer$Visitor.process(ExpressionAnalyzer.java:427)
	at io.trino.sql.analyzer.ExpressionAnalyzer.analyze(ExpressionAnalyzer.java:359)
	at io.trino.sql.analyzer.ExpressionAnalyzer.analyzeExpression(ExpressionAnalyzer.java:1973)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.analyzeExpression(StatementAnalyzer.java:2925)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.analyzeWhere(StatementAnalyzer.java:2758)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.lambda$visitDelete$7(StatementAnalyzer.java:640)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitDelete(StatementAnalyzer.java:640)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitDelete(StatementAnalyzer.java:343)
	at io.trino.sql.tree.Delete.accept(Delete.java:61)
	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.process(StatementAnalyzer.java:360)
	at io.trino.sql.analyzer.StatementAnalyzer.analyze(StatementAnalyzer.java:323)
	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:91)
	at io.trino.sql.analyzer.Analyzer.analyze(Analyzer.java:83)
	at io.trino.execution.SqlQueryExecution.analyze(SqlQueryExecution.java:263)
	at io.trino.execution.SqlQueryExecution.<init>(SqlQueryExecution.java:186)
	at io.trino.execution.SqlQueryExecution$SqlQueryExecutionFactory.createQueryExecution(SqlQueryExecution.java:768)
	at io.trino.dispatcher.LocalDispatchQueryFactory.lambda$createDispatchQuery$0(LocalDispatchQueryFactory.java:129)
	at io.trino.$gen.Trino_testversion____20210317_095608_2849.call(Unknown Source)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

### PR DESCRIPTION
**Pull request by Praveen2112**
Fixes #7317

### CODE REVIEW

**Review by ksobolew — APPROVED** [review]
Down with flaky tests!

**Review by kokosing — APPROVED** [review]

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java [hunk 1]
```diff
@@ -65,51 +66,60 @@ public void testCreateTableWithDecimalColumn()
...
 
-        assertUpdate("INSERT INTO test_decimal VALUES (0, 0.0), (2, 2.2), (1, 1.1)", 3);
-        assertQuery("SELECT * FROM test_decimal WHERE col_decimal = 1.1", "VALUES (1, 1.1)");
-        assertUpdate("DELETE FROM test_decimal WHERE col_decimal = 1.1", 1);
+        assertUpdate(format("INSERT INTO %s VALUES (0, 0.0), (2, 2.2), (1, 1.1)", tableName), 3);
+        assertQuery(format("SELECT * FROM %s WHERE col_decimal = 1.1", tableName), "VALUES (1, 1.1)");
+        assertUpdate(format("DELETE FROM %s WHERE col_decimal = 1.1", tableName), 1);
 
-        assertUpdate("DROP TABLE test_decimal");
+        assertUpdate(format("DROP TABLE %s", tableName));
```
[the remark this answers is not in the document]
Comment (Praveen2112) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java]: Have converted it to `TestTable`.

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java [hunk 2]
```diff
@@ -65,70 +68,53 @@ public void testCreateTableWithDecimalColumn()
...
-
-        assertUpdate("DROP TABLE test_decimal");
+        try (TestTable testTable = new TestTable(
+                new TrinoSqlExecutor(getQueryRunner()),
+                "test_decimal",
+                "(id INT WITH (primary_key=true), col_decimal decimal(10, 6)) " +
+                        "WITH (partition_by_hash_columns = ARRAY['id'], partition_by_hash_buckets = 2)")) {
+            assertUpdate(format("INSERT INTO %s VALUES (0, 0.0), (2, 2.2), (1, 1.1)", testTable.getName()), 3);
+            assertQuery(format("SELECT * FROM %s WHERE col_decimal = 1.1", testTable.getName()), "VALUES (1, 1.1)");
+            assertUpdate(format("DELETE FROM %s WHERE col_decimal = 1.1", testTable.getName()), 1);
```
Comment (kokosing) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java:78]: Run select after to see if proper record was actually removed

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java [hunk 3]
```diff
@@ -65,70 +68,53 @@ public void testCreateTableWithDecimalColumn()
...
-        assertQuery("SELECT * FROM test_decimal WHERE col_decimal = 1.1", "VALUES (1, 1.1)");
-        assertUpdate("DELETE FROM test_decimal WHERE col_decimal = 1.1", 1);
-
-        assertUpdate("DROP TABLE test_decimal");
+        try (TestTable testTable = new TestTable(
+                new TrinoSqlExecutor(getQueryRunner()),
+                "test_decimal",
+                "(id INT WITH (primary_key=true), col_decimal decimal(10, 6)) " +
+                        "WITH (partition_by_hash_columns = ARRAY['id'], partition_by_hash_buckets = 2)")) {
+            assertUpdate(format("INSERT INTO %s VALUES (0, 0.0), (2, 2.2), (1, 1.1)", testTable.getName()), 3);
```
Comment (kokosing) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java:76]: TestTable allows to insert data.

**Review by kokosing — APPROVED** [review]

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java (hunk 3 above)
Comment (Praveen2112) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java:76]: But it doesn't validates the number of rows inserted.

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java [hunk 4]
```diff
@@ -144,7 +131,7 @@ private void doTestCreateTableWithDecimalColumn(TestDecimal decimal)
 
         String getTableName()
         {
-            return format("test_decimal_%s_%s", precision, scale);
+            return format("test_decimal_%s_%s_%s", randomTableSuffix(), precision, scale);
```
Comment (ksobolew) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java:134]: Isn't this redundant now? `TestTable` will add yet another suffix

**Review by ksobolew — APPROVED** [review]

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java (hunk 4 above)
Comment (Praveen2112) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduIntegrationDecimalColumns.java:134]: Yes !! Missed it

**Pull request merged**